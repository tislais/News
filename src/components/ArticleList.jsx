import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';

const ArticleList = ({ articles }) => {
  const articleMap = articles.map((article) => (
    <li key={article.title}>
      <Article
        title={article.title}
        author={article.author}
        description={article.description} />
    </li>
  ));
  return articleMap;
};

ArticleList.propTypes = {
  articles: PropTypes.objectOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      author: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
    })
  ).isRequired
};

export default ArticleList;
