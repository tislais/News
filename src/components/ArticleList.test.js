import React from 'react';
import { render } from '@testing-library/react';
import ArticleList from './ArticleList';

describe('ArticleList component', () => {
  it('renders title, author, description', () => {
    const articles = [{
      title: 'title',
      author: 'author',
      description: 'description'
    }];

    const { asFragment } = render(
      <ArticleList articles={articles}/>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
