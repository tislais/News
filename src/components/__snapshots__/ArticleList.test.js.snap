// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList component renders title, author, description 1`] = `
<DocumentFragment>
  <li>
    <h1>
      title
    </h1>
    <p>
      author
    </p>
    <p>
      description
    </p>
  </li>
</DocumentFragment>
`;
